c/logging.c

```diff
@@ -439,47 +444,219 @@ LoggingDestination *logConfigureDestination(LoggingContext *context,
+
+  return result;
+}
+
+static const char *getServiceName() {
+  return "ZWESZ1";
+}
+
+static void getTaskInformation(char *taskInformation, unsigned int taskInformationSize) {
+  pthread_t pThread = pthread_self();
+  snprintf(taskInformation, taskInformationSize, "0x%p:0x%p:%d", getTCB(), &pThread, getpid());
+}
+
+static void getMessage(char *message, unsigned int messageSize, const char *formatString, va_list argList) {
+  vsnprintf(message, messageSize, formatString, argList);
+}
+
+/*
+ * 1. Try to get the name from the TCB.
+ * 2. Try to get the name from the login data base.
+ */
+static void getUserID(char *user, unsigned int userSize) {
+  TCB *tcb = getTCB();
+  ACEE *acee = (ACEE *) tcb->tcbsenv;
+  if (acee) {
+    snprintf(user, userSize, "%.*s", acee->aceeuser[0], acee->aceeuser[1]);
+  } else {
+    getlogin_r(user, userSize);
+  }
+}
+
+/*
+ * These seem very generous.
+ *
+ */
+#define LOG_USER_ID_MAX_LENGTH   8
+#define LOG_MESSAGE_MAX_LENGTH   512
+#define LOG_TASK_INFO_MAX_LENGTH 128
+
+static void prependMetadata(int logLevel,
+                            char *fullMessage,
+                            unsigned int fullMessageSize,
+                            char *formatString,
+                            va_list argList,
+                            char *fileName,
+                            char *functionName,
+                            unsigned int lineNumber) {
+  char user[LOG_USER_ID_MAX_LENGTH + 1] = {0};
+  getUserID((char *) &user, sizeof(user) - 1);
+  zl_time_t time = {0};
+  time = gettime();
+  char message[LOG_MESSAGE_MAX_LENGTH + 1] = {0};
+  getMessage((char *) &message, sizeof(message), formatString, argList);
+  char taskInformation[LOG_TASK_INFO_MAX_LENGTH + 1] = {0};
+  getTaskInformation((char *) &taskInformation, sizeof(taskInformation));
+  char *logLevelAsString = logLevelToString(logLevel);
+  /*
+   * The following fields were added in the second iteration of LogHandler.
+   *
+   * 1. fileName
+   * 2. functionName
+   * 3. lineNumber
+   * 4. logLevel
+   *
+   * (1,2,3) were never asked for, but (4) wasn't propogated down to the LogHandler and was just used to determine
+   * if the message should be printed. (4) is now passed to here so that it can be used to construct the common
+   * logging format.
+   *
+   * If one of these (1,2,3,4) isn't present, then they all aren't based on this implementation; therefore, we'll just format with what's
+   * available to us without relying on new arguments. The previous LogHandler will return null or empty values for (1,2), a 0 for (3), and
+   * ZOWE_LOG_NA for (4), which when going thorugh logLevelToString() will be NA.
+   *
+   * The existing logs in the server should contain the level, as long as logConfigureDestination3() is used. If not, it will have the placeholder.
+   *
+   */
+  if (!fileName || strlen(fileName) == 0 || !functionName || strlen(functionName) == 0 || lineNumber == 0) {
```

Comment:
Could you factor this check out and call it something meaningful?

For example (you may want to change the name to something more fitting):
```c
bool newFormat = fileName && strlen(fileName) != 0 && functionName &&
                 strlen(functionName) != 0 && lineNumber != 0;
```